python/ray/serve/_private/application_state.py

```diff
@@ -182,6 +187,9 @@ def update(self):
                     self.deploy_obj_ref = None
                     logger.warning(self.app_msg)
                     return
+
+            self._process_terminating_deployments()
```

Comment:
why is this moved? if the placement is important, please leave a comment indicating it

Reply:
That was a mistake, reverted